ketcher-autotests/tests/Macromolecule-editor/Macro-Micro-Switcher/macro-micro-switcher.spec.ts

```diff
@@ -0,0 +1,439 @@
+    .getByText('★')
+    .click();
+  await page.getByTestId('meM___N-Methyl-Methionine').getByText('★').click();
+  await page.getByTestId('RNA-TAB').click();
+  await page.getByTestId('summary-Sugars').click();
+  await page.getByTestId('25R___2,5-Ribose').getByText('★').click();
+  await page.getByTestId('summary-Bases').click();
+  await page.getByTestId('baA___N-benzyl-adenine').getByText('★').click();
+  await page.getByTestId('summary-Phosphates').click();
+  await page.getByTestId('bP___Boranophosphate').getByText('★').click();
+  await page.getByTestId('CHEM-TAB').click();
+  await page.getByTestId('Test-6-Ch___Test-6-AP-Chem').getByText('★').click();
+}
+
+test.describe('Macro-Micro-Switcher', () => {
+  test.beforeEach(async ({ page }) => {
+    await waitForPageInit(page);
+    await turnOnMacromoleculesEditor(page);
+  });
+
+  test('Check that preview window of macro structure doesn not change in micro mode ', async ({
+    page,
+  }) => {
+    /* 
+    Test case: https://github.com/epam/ketcher/issues/3603
+    Description: Preview window of macro structure doesn't change in micro mode
+    Test working incorrect now because we have bug https://github.com/epam/ketcher/issues/3603
+    */
+    const moleculeLabels = ['A', '25R', 'baA', 'Test-6-Ph', 'Test-6-Ch'];
+    await openFileAndAddToCanvas('KET/five-monomers.ket', page);
+    await turnOnMicromoleculesEditor(page);
+    for (const label of moleculeLabels) {
+      await page.getByText(label, { exact: true }).hover();
+      await takeEditorScreenshot(page);
+    }
+  });
+
+  test('Check that macromolecule structures in micromode are represented as S-Groups with bonds', async ({
+    page,
+  }) => {
+    /* 
+    Test case: Macro-Micro-Switcher
+    Description: Macromolecule structures in micromode are represented as S-Groups with bonds
+    */
+    await openFileAndAddToCanvas(
+      'KET/three-monomers-connected-with-bonds.ket',
+      page,
+    );
+    await turnOnMicromoleculesEditor(page);
+    await page.getByText('Edc').hover();
+    await takeEditorScreenshot(page);
+  });
+
+  test('Micromolecules in macromode will be represented as CHEMs with generated name(F1, F2, ...Fn)', async ({
+    page,
+  }) => {
+    /* 
+    Test case: Macro-Micro-Switcher
+    Description: Micromolecules in macromode represented as CHEMs with generated name(F1, F2, ...Fn)
+    */
+    await turnOnMicromoleculesEditor(page);
+    await openFileAndAddToCanvas('KET/eight-micromolecules.ket', page);
+    await turnOnMacromoleculesEditor(page);
+    await takeEditorScreenshot(page);
+  });
+
+  test('Check that After hiding Library in Macro mode possible to see Show Library button', async ({
+    page,
+  }) => {
+    /* 
+    Test case: Macro-Micro-Switcher
+    Description: After hiding Library in Macro mode 'Show Library' button is visible.
+    */
+    await page.getByText('Hide').click();
+    await takePageScreenshot(page);
+    await page.getByText('Show Library').click();
```

Comment:
Maybe it would be worth to add check: await page.getByText('Show Library').isVisible(); ?

Reply:
Good point) But I trust the screenshot more)

Reply:
We can keep both verifications ;)